src/gp/src/TensorProductMesh.C

```diff
@@ -273,7 +275,21 @@ TensorProductMesh<V>::generateDiscrepancyBases
         }
     }
 
-  queso_not_implemented(); // FIXME
+  // Finally we need to normalize the bases we've just generated
+  double maxnorm = 0;
+  for (unsigned int b=0; b != bases.size(); ++b)
+    {
+      const V & basis = *bases[b];
+      maxnorm = std::max(maxnorm, basis.norm2());
+    }
+
+  const double invmaxnorm = 1./maxnorm;
+
+  for (unsigned int b=0; b != bases.size(); ++b)
+    {
+      V & basis = *bases[b];
+      basis *= invmaxnorm;
+    }
```

Comment:
Hmmm, I left a couple of comments pointing out you're missing the normalisation constant.  I take it those comments are now irrelevant.

Reply:
In theory, yes.

In practice those comments may have completely vanished, because I get "Sometimes commits can disappear after a force-push." messages when trying to look at them in context...